specification.markdown

```diff
@@ -313,35 +313,47 @@ The `name` provides the official human-readable name of the `EBML Element`. The
 
 The `name` attribute is REQUIRED.
 
-#### level
+#### path
 
-The `level` notes at what hierarchical depth or depths the `EBML Element` MUST occur if used within an `EBML Document`. The `Root Element` of an `EBML Document` is at Level 0 and the `Child Elements` that it contains are at Level 1. The `level` MUST be expressed as an integer. `EBML Elements` defined as `global` and `recursive` MAY occur at a level greater than or equal to the defined `level`.
+The path of the `EBML Element`. This path MUST be defined with the full hierarchy of `EBML Elements` separated with a `/`. The top `EBML Element` in the path hierarchy being the first in the value. The syntax of the `path` attribute is defined using this Extended Backus-Naur Form (EBNF) [@!ISO.14977.1996] notation:
 
-The `level` attribute is REQUIRED.
+```
+EBMLFullPath          ::= EBMLFullPath, [EBMLElementOccurence]
+EBMLReferencePath     ::= EBMLParentPath, EBMLElementPath
+EBMLParentPath        ::= (EBMLPathAtom)* | AllWildcard
+EBMLElementPath       ::= EBMLPathAtom | EBMLPathAtomRecursive
+EBMLPathAtom          ::= "/", EBMLAtomName
+EBMLPathAtomRecursive ::= "(", EBMLPathAtom, ")", "+"
+EBMLAtomName          ::= (EBMLNameChar)+ | AllWildcard | AtLeastOnce
+EBMLNameChar          ::= [A-Z] | [a-z] | [0-9] | "-" | "."
+AllWildcard           ::= "*"
+AtLeastOnce           ::= "+"
+EBMLElementOccurence  ::= "(", [MinOccurence], ",", [MaxOccurence] ")" 
+MinOccurence          ::= [0-9]+
+MaxOccurence          ::= [0-9]+ | "unbounded"
```

Comment:
The `name` attribute is defined as a valid `NCName` but here it implies that EBMLNameChar is `[A-Z] | [a-z] | [0-9] | "-" | "."`. These should be synced, so that `name` is also constrained to `[A-Z] | [a-z] | [0-9] | "-" | "."`. Also it should assert somewhere that `name` and `EBMLAtomName` MUST be the same.